custom_components/octopus_energy/api_client_home_pro/__init__.py

```diff
@@ -73,7 +73,7 @@ async def async_get_consumption(self, is_electricity: bool) -> list | None:
             divisor = int(data["raw"]["divisor"], 16)
             return [{
               "total_consumption": int(data["consumption"]) / divisor if divisor > 0 else None,
-              "demand": float(data["instdmand"]) if "instdmand" in data else None,
+              "demand": float(data["instdmand"]) / (divisor / 1000) if divisor > 0 and "instdmand" in data else None,
```

Comment:
question: Why are you dividing the divisor by 1000?

Reply:
Hi, with the 10000 divisor, you get demand in kW.
Hence the 1000.

Reply:
Sorry I don't think I'm following. The sensor reports the data in watts not kilowatts. And the docs state this API should also report the data in watts.

I don't have an issue with this sensor, but understand others do. My divisor is 1000, so with this calculation, it makes no adjustment on my setup (which is correct).

Are you saying your divisor is reporting 10000 and it's out by a factor of 10 in your instance? If so, what's your multiplier value?

Reply:
https://forum.octopus.energy/t/for-the-pro-user/8453/2892?u=justpassing

I hope this is clearer. Happy new year and thanks for all your great work.